pandas/core/arrays/string_arrow.py

```diff
@@ -0,0 +1,628 @@
+        """
+
+        value, method = validate_fillna_kwargs(value, method)
+
+        mask = self.isna()
+
+        if is_array_like(value):
+            if len(value) != len(self):
+                raise ValueError(
+                    f"Length of 'value' does not match. Got ({len(value)}) "
+                    f"expected {len(self)}"
+                )
+            value = value[mask]
+
+        if mask.any():
+            if method is not None:
+                func = get_fill_func(method)
+                new_values = func(self.astype(object), limit=limit, mask=mask)
+                new_values = self._from_sequence(new_values)
+            else:
+                # fill with value
+                new_values = self.copy()
+                new_values[mask] = value
+        else:
+            new_values = self.copy()
+        return new_values
+
+    def _reduce(self, name, skipna=True, **kwargs):
+        if name in ["min", "max"]:
+            return getattr(self, name)(skipna=skipna)
+
+        raise TypeError(f"Cannot perform reduction '{name}' with string dtype")
+
+    @property
+    def nbytes(self) -> int:
+        """
+        The number of bytes needed to store this object in memory.
+        """
+        return self.data.nbytes
+
+    def isna(self) -> np.ndarray:
+        """
+        Boolean NumPy array indicating if each value is missing.
+
+        This should return a 1-D array the same length as 'self'.
+        """
+        # TODO: Implement .to_numpy for ChunkedArray
+        return self.data.is_null().to_pandas().values
+
+    def copy(self) -> ExtensionArray:
+        """
+        Return a copy of the array.
+
+        Parameters
+        ----------
+        deep : bool, default False
+            Also copy the underlying data backing this array.
+
+        Returns
+        -------
+        ExtensionArray
+        """
+        return type(self)(self.data)
+
+    def _cmp_method(self, other, op):
+        from pandas.arrays import BooleanArray
+
+        ops = {
+            "eq": pc.equal,
+            "ne": pc.not_equal,
+            "lt": pc.less,
+            "gt": pc.greater,
+            "le": pc.less_equal,
+            "ge": pc.greater_equal,
+        }
+        pc_func = ops[op.__name__]
+        if isinstance(other, (ABCSeries, ABCDataFrame, ABCIndex)):
+            return NotImplemented
+        if isinstance(other, ArrowStringArray):
+            result = pc_func(self.data, other.data)
+        elif is_scalar(other):
+            result = pc_func(self.data, pa.scalar(other))
+        else:
+            rops = {
+                "eq": operator.eq,
+                "ne": operator.ne,
+                "lt": operator.gt,
+                "gt": operator.lt,
+                "le": operator.ge,
+                "ge": operator.le,
+            }
+            rop = rops[op.__name__]
+            result = rop(other, self)
+            return BooleanArray._from_sequence(result)
```

Comment:
Should we return `NotImplemented` instead? (and eg explicitly check for StringArray and convert that to ArrowStringArray first)

Reply:
I changed this in https://github.com/pandas-dev/pandas/pull/35259/commits/f651563571074af5d9cd8ed2f572431e1eaf3daa since the results for test_compare_array with a numpy object array were correct with NotImplemented (i.e. allowing the reversed op) and just the dtype was incorrect (bool instead of boolean).

I admit that this was a hacky solution and likely to cause bugs with comparison with other types.

will change to NotImplemented to allow most types for fall through to the reversed op and look for a better solution for explicit comparison with numpy arrays so that a BooleanArray is returned.

Reply:
added a explicit check for numpy arrays to get the base extension tests passing. probably need a deeper discussion (and additional tests) on whether we dispatch to pyarrow for other types.